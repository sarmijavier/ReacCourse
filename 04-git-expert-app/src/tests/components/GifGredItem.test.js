import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';


import { GifGridItem } from '../../components/GifGridItem';

const title = 'un titulo';
const url = 'https://localhost/algo.jpg';
const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )


describe('Pruebas en <GifGridItem >', () => {

    test('Debería de mostrar <PrimeraApp /> correctament ', () => {
        

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de tener un parráfo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('Debe de tener la imagen igual al urly alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log( img.html() )
        //console.log( img.props().src )
        expect( img.prop('src') ).toBe( url )

    })

    test('Debe de tener animate__fadeInLeftBig', () => {
        
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect( className.includes('animate__fadeInLeftBig') ).toBe( true )
        //const cadena = (div.props().className).split(' ')
        //expect( cadena[2] ).toBe( 'animate__fadeInLeftBig' )

    })
    
    
    
})
 
