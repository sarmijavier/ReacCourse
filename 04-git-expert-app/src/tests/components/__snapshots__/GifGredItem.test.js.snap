// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem > Deber√≠a de mostrar <PrimeraApp /> correctament  1`] = `
<div
  className="card animate__animated animate__fadeInLeftBig"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
